lambda/MAS/Controllers/MailController.cs

```diff
@@ -110,5 +113,66 @@ public async Task<IActionResult> PutMailAsync(DateTime? date = null)
                 return StatusCode(500, new ProblemDetails { Status = 500, Title = e.Message, Detail = e.InnerException?.Message });
             }
         }
+
+        //PUT api/mail/weekly or api/mail/weekly/2020-01-13
+        [HttpPut("weekly/{date?}")]
+        public async Task<IActionResult> PutWeeklyMailAsync([FromRoute] DateTime? date = null)
+        {
+            Weekly weeklyContent;
+
+            var todaysDate = date ?? DateTime.Today;
+            var isBankHoliday = _bankHolidayService.IsBankHoliday(todaysDate);
+            if (todaysDate.DayOfWeek == DayOfWeek.Monday)
+            {
+                if (isBankHoliday)
+                {
+                    _logger.LogWarning($"{todaysDate.ToString("dd/MM/yyyy")} is a bank holiday therefore an email isnt sent");
+                    return Content($"{todaysDate.ToString("dd/MM/yyyy")} is a bank holiday therefore an email isnt sent");
+                }
+                else
+                {
+                    weeklyContent = await _contentService.GetWeeklyAsync(todaysDate);
+                }
+            }
+            else
+            {
+                var previousMonday = todaysDate.AddDays(-(int)todaysDate.DayOfWeek + 1);
+                var previousMondayIsBankHoliday = _bankHolidayService.IsBankHoliday(previousMonday);
+                if (previousMondayIsBankHoliday)
+                {
+                    weeklyContent = await _contentService.GetWeeklyAsync(previousMonday);
+                }
+                else
+                {
+                    _logger.LogWarning($"An email was sent on {previousMonday.ToString("dd/MM/yyyy")}");
+                    return Content($"An email was sent on {previousMonday.ToString("dd/MM/yyyy")}");
+                }
+            }
+
+            if (weeklyContent == null)
+            {
+                _logger.LogWarning("No weekly record was found");
+                return Content("No weekly record was found");
+            }
+            else if (weeklyContent.Items.Count == 0)
+            {
+                _logger.LogWarning("The weekly didn't have any items");
+                return Content("The weekly didn't have any items");
+            }
+
+            var body = await _viewRenderer.RenderViewAsync(this, "~/Views/Mail/Weekly.cshtml", weeklyContent);
+            var previewText = "NICE Medicines Awareness Weekly. A selection of the week's current awareness and evidence-based medicines information";
+
+            try
+            {
+                var campaign = await _mailService.CreateAndSendWeeklyAsync(previewText, body, weeklyContent.Title);
+                return Json(campaign);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Failed to create and send weekly campaign - exception: {e.Message}");
```

Comment:
Can you pass this exception to the logger so we have the full exception in Kibana?